### DISCUSSION TITLE
Support for ulid rkeys

### DISCUSSION THREAD

**Discussion by ngerakines** [bluesky-social/atproto#2690] [Protocol (atproto)]
I'm interested in using https://github.com/ulid/spec as rkeys. Is there interest in using ulids in the future?

**Comment by DavidBuchanan314** [comment]
rkeys don't need to be globally unique, just unique within a collection within a repo. But, there's nothing stopping you from using an ulid as a rkey today, if you wanted to.

**Comment by bnewbold** [comment]
As Dave says, sure you can just go ahead an use them today, using the `any` rkey format.

Was thinking that we might want to support a specific UUID-based rkey format and/or Lexicon string format, for folks who are UUID-pilled. We'll probably continue to endorse TIDs as the most atproto-native option, but UUIDs are a totally fine choice for new work, and there are also probably use-cases where they are necessary or preferred (embedding existing content identifiers, using specific datastores or sequencers, or applications where the rkey is expected to be more universally unique, though of course can't stop anybody from intentionally colliding in a different repo).

There is bike-shedding around the best UUID encoding... most of atproto is lower-case default, and record keys are case-sensitive. ULIDs seem mostly fine though.
